inspector/src/modules/bitrate_estimate_module.ts

```diff
@@ -0,0 +1,338 @@
+      canvasCtx.beginPath();
+      canvasCtx.strokeStyle = "lightgrey";
+      canvasCtx.lineWidth = 1;
+      height = canvasElt.height - HEIGHT_MARGIN_TOP - HEIGHT_MARGIN_BOTTOM;
+      let nbGridLines;
+      if (height > 300) {
+        nbGridLines = 10;
+      } else if (height > 200) {
+        nbGridLines = 7;
+      } else if (height > 100) {
+        nbGridLines = 5;
+      } else if (height > 50) {
+        nbGridLines = 3;
+      } else {
+        nbGridLines = 2;
+      }
+      const stepHeight = height / nbGridLines;
+      const stepVal = currentMaxBitrate / nbGridLines;
+      for (let i = 0; i <= nbGridLines; i++) {
+        const nHeight = stepHeight * i + HEIGHT_MARGIN_TOP;
+        canvasCtx.moveTo(0, nHeight);
+        canvasCtx.font = "14px Arial";
+        const currStepVal = stepVal * (nbGridLines - i);
+        const isDark =
+          configState.getCurrentState(STATE_PROPS.CSS_MODE) === "dark";
+        if (isDark) {
+          canvasCtx.fillStyle = "#ffffff";
+        } else {
+          canvasCtx.fillStyle = "#000000";
+        }
+        canvasCtx.fillText(convertToReadableBitUnit(currStepVal), 0, nHeight);
+        canvasCtx.lineTo(canvasElt.width, nHeight);
+      }
+      canvasCtx.stroke();
+    }
+
+    /**
+     * Draw all data contained in `data` in the canvas given.
+     */
+    function drawData(): void {
+      if (canvasCtx === null) {
+        return;
+      }
+      canvasCtx.beginPath();
+      canvasCtx.strokeStyle = "rgb(200, 100, 200)";
+      canvasCtx.lineWidth = 2;
+      canvasCtx.moveTo(0, bitrateValueToY(data[0].bitrateEstimate ?? 0));
+      for (let i = 1; i < data.length; i++) {
+        canvasCtx.lineTo(
+          dateToX(data[i].timestamp),
+          bitrateValueToY(data[i].bitrateEstimate ?? 0),
+        );
+      }
+      canvasCtx.stroke();
+
+      // add measure points
+      canvasCtx.fillStyle = "rgb(200, 100, 200)";
+      for (let i = 1; i < data.length; i++) {
+        const x = dateToX(data[i].timestamp);
+        const y = bitrateValueToY(data[i].bitrateEstimate ?? 0);
+        const circle = new Path2D();
+        circle.arc(x, y ?? 0, 4, 0, 2 * Math.PI);
+        canvasCtx.fill(circle);
+
+        const tooltip = strHtml`<span> </span>`;
+        tooltip.innerText = convertToReadableBitUnit(
+          data[i].bitrateEstimate ?? 0,
+        );
+        tooltip.style.position = "absolute";
+        tooltip.style.visibility = "hidden";
+        // center the tooltip and display it just a bit higher than the point
+        tooltip.style.top = `${y - 30}px`;
+        tooltip.style.left = `${x - 20}px`;
+        canvasOverlay.appendChild(tooltip);
+        const eventHander = (e: MouseEvent) => {
+          if (canvasCtx.isPointInPath(circle, e.offsetX, e.offsetY)) {
+            tooltip.style.visibility = "visible";
+          } else {
+            tooltip.style.visibility = "hidden";
+          }
+        };
+        canvasElt.addEventListener("mousemove", eventHander);
```

Comment:
I've seen an issue of multiple superposing texts if they share a very close point :/

How should we handle this? With a background at worse?